packages/nimble-components/src/table/specs/cell-state-when-scrolling-hld.md

```diff
@@ -0,0 +1,98 @@
+    -   Chrome/Safari: Works when the text selection is fully within the table
+    -   Chrome: Works when the text selection is partially in the table, partially outside the table
+    -   Safari: Doesn't work when the text selection is partially in the table, partially outside the table. In that case, the `compareBoundaryPoints` results match what we'd get if the text selection was only before / only after the table (unusable).
+    -   Firefox: Errors out with a `WrongDocumentError` (i.e. one range is in a shadow root, and the other isn't / is in a different shadow root)
+-   (Chrome only) Look at `table.shadowRoot.getSelection()`. If non-null and `rangeCount > 0` we have selected text somewhere in the table. Simplest approach, but this API is non-standard / not in the other browsers / may be removed at any time in the future.
+
+**Implementation Plan:**  
+Based on the observations above, the current plan to check if the table contains selected text is:  
+Get `window.getSelection()`. If null or `rangeCount === 0`, no text is selected. Otherwise, for each `Range`:
+
+-   Check if `startContainer`/`endContainer` is the `nimble-table`
+-   Else, do the `compareBoundaryPoints` check (in a `try/catch`)
+-   Else, check if `startContainer`/`endContainer` has the `nimble-table` as an ancestor
+
+If any of those `Range` checks succeed, remove that `Range` from the selection.  
+**Limitations:**
+
+-   If text is selected in the table but outside rows (e.g. column header text), it will also be cleared when the user scrolls. There's not a good way to differentiate the text location that works in each browser, so that seems like an acceptable compromise.
+-   (Safari only) If text selection is partially in the table and partially before/ after it, there isn't any way for us to detect that case. So Safari will still have text incorrectly selected after the scroll in that case.
+
+**Planned API:**  
+This logic will be in the Nimble Virtualizer class, called from `handleVirtualizerChange()` (called when the user scrolls). It will apply to all Nimble tables / all column types, without an opt-out option.
+
+### Blur Focused Controls in Cells
+
+**Implementation Plan:**  
+Logic to find if the table contains an active/focused element within a cell:  
+Start with `document.activeElement`. If non-null, check if that element is a shadow root. If so, continue recursively with `shadowRoot.activeElement`. Stop when we reach a `nimble-table-cell` or `null`.
+
+If we have a focused cell:
+
+-   If the cell has an action menu (note: in this case the focused element will be a `nimble-menu-item`), find the associated `nimble-menu-button`. If open, set `open` to `false`.
+-   Get the `TableColumn` instance for the column that the cell is in, and call the new API `onBeforeFocusedCellRecycled(cell: TableCell)` on it (if defined).
```

Comment:
I guess I don't have the clearest picture of why this abstract method would be warranted (and ultimately, this feels like a bit of an awkward concern to hoist upon the column API). If I understand the flow of things, by the time you would have called this abstract method we will have already discovered the element that is the `document.activeElement` (and whether it is in a `nimble-table-cell`). At this point, couldn't we just dumbly call `blur()` on that element on scroll? What would this do for menu components that are in an open state? I would think for non-input components, this would result in a no-op, or at least nothing of consequence (I could be wrong here).

Reply:
If we find the leafmost active element (instead of stopping at `nimble-table-cell`), we'll stop at the `input` or `textarea` inside the `nimble-text-field`/`nimble-number-field`/`nimble-text-area`. Calling `blur()` on those should work, and be sufficient.
Calling `blur()` on a `nimble-menu-item` is fine (it doesn't throw an error and it does lose focus), but I saw some cases where the menu / menu button remained open after doing that. But I suppose we could still have custom code to close menus if needed directly in the virtualizer without adding the `onBeforeFocusedCellRecycled` API.

Reply:
Current plan is to not have a default implementation of `onBeforeFocusedCellRecycled` (still have a note about it in Open Issues if there's any disagreement on that).

The proposal @rajsite outlined above about firing a custom event on the cell (cell's content container specifically, based on our chat) is captured in the HLD as "Option 1" in the section about blurring cells. The original `onBeforeFocusedCellRecycled` proposal is Option 2.  

Added an Open Issue to choose between those 2 proposals.